cider/exec/nextgen/Nextgen.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "exec/nextgen/Nextgen.h"
+
+namespace cider::exec::nextgen {
+
+std::unique_ptr<context::CodegenContext> compile(const RelAlgExecutionUnit& ra_exe_unit,
+                                                 const jitlib::CompilationOptions& co) {
+  auto context = std::make_unique<context::CodegenContext>();
+  jitlib::LLVMJITModule module("codegen", true, co);
```

Comment:
Q: In LLVM, can we use same name for a module "codegen" if we compile multiple query fragement in a single process?

Reply:
It should ok since name only as a identifier for debugging or logging